### PR TITLE
[#1m1gdaf] add GamePage

### PR DESCRIPTION
**Pull request by VladimirAfanasievFS**
# Summary

#### A brief description of the pull request.

[Task](https://app.clickup.com/t/1m1gdaf)

#### Screenshots in case of UI changes

# Test plan

#### List of steps to manually test introduced functionality:

- Go to [ReviewApp](https://facegame-staging-pr-6.herokuapp.com)
- ...

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### CODE REVIEW

File: components/pages/result/ResultPage.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { useRouter } from 'next/router';
+import { Title, PageContent, Content, PreviewImg, ImgGroup } from './styled';
+
+const ResultPage = () => {
+  const router = useRouter();
+  const handleClick = () => {
+    router.push(GAME);
+  };
+  return (
+    <NotifierProvider>
```
Comment (VladimirAfanasievFS) [inline components/pages/result/ResultPage.js:18]: Type error: Property 'accessTokenManager' is missing in type '{ children: Element; }' but required in type '{ [x: string]: any; accessTokenManager: any; }'.

Вот тут не понимаю как вызывать компоненты из не ts файлов, чтобы не ругался

File: components/pages/index/HomePage.js [hunk 2]
```diff
@@ -41,7 +47,7 @@ const HomePage = () => {
               speed.
             </TagLine>
             {user ? (
-              <button type="button" style={{ width: '200px', height: '60px' }}>
+              <button type="button" style={{ width: '200px', height: '60px' }} onClick={handleClick}>
```
Comment (nadezhdakharchuk) [inline components/pages/index/HomePage.js:50]: It should be better to use <Link> for changing route. You can use `ButtonedLink` component with custom styles
https://nextjs.org/docs/api-reference/next/link

```
import Link from 'next/link';
import ButtonedLink from 'components/shared/atoms/ButtonedLink';

<Link href={GAME} passHref>
  <ButtonedLink>начать новую игру</ButtonedLink>
</Link>

```

File: components/pages/result/ResultPage.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          <button type="button" style={{ width: '200px', height: '60px' }} onClick={handleClick}>
+            начать новую игру
+          </button>
+        </PageContent>
+      </DefaultTemplate>
+    </NotifierProvider>
+  );
+};
+
+export default withApolloClient(WithAuth(ResultPage));
```
Comment (nadezhdakharchuk) [inline components/pages/result/ResultPage.js:36]: Please wrap this page with `WithAuthSecurity`

```
import WithAuth from 'lib/auth/withAuth';
import { withApolloClient } from 'lib/withApolloClient';
import WithAuthSecurity from 'lib/auth/withAuthSecurity';

export default withApolloClient(WithAuth(WithAuthSecurity(ResultPage)));
```

File: components/pages/game/GamePage.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  return (
+    <NotifierProvider>
+      <DefaultTemplate>
+        {currentQuestion && <GameStep question={currentQuestion} addAnswer={addAnswer(currentQuestion)} />}
+      </DefaultTemplate>
+    </NotifierProvider>
+  );
+};
+
+export default withApolloClient(WithAuth(GamePage));
```
Comment (nadezhdakharchuk) [inline components/pages/game/GamePage.tsx:128]: Please wrap this page with `WithAuthSecurity`

```
import WithAuth from 'lib/auth/withAuth';
import { withApolloClient } from 'lib/withApolloClient';
import WithAuthSecurity from 'lib/auth/withAuthSecurity';

export default withApolloClient(WithAuth(WithAuthSecurity(GamePage)));
```

File: components/pages/result/ResultPage.js (hunk 1 above)
Comment (nadezhdakharchuk) [reply components/pages/result/ResultPage.js:18]: Screenshot 2021-10-12 at 09 31 25
Let's rename GamePage.tsx to GamePage.js for now and try to fix this bug with FE team in the `react-next-base`

And please move `GameStep` to component file and logic helpers to separate component

**Pull request merged**